File: frontend/library/lib/courses/curriculum/curriculum.ts
import { Tutorial, db_to_tutorial } from "../tutorials/tutorial";
import {
    db,
    auth
} from '/lib/firebase/initFirebase';

import {
    collection,
    doc,
    getDocs,
    setDoc,
    deleteDoc,
    query,
    where
} from "firebase/firestore";


export type TutorialCurriculum =  Tutorial & {
    sessionNumber: number;
};



export type Curriculum = {
    id: string;
    title: string;
    isHolidayProgram: boolean;
    numSessions: number;
    holidayModule: any;
    tutorials: Tutorial[];
    additionalInfo: string;
};

File: backend/functions/lib/programs/docs/curriculum/curriculum.py



from dataclasses import dataclass, field
from typing import List, Any


from lib.programs.docs.tutorials import Tutorial



@dataclass
class Curriculum:
    id: str
    title: str
    is_holiday_program: bool
    num_sessions: int
    holiday_module: Any
    tutorials: List[Tutorial] = field(default_factory=list)
    additional_info: str